### PR TITLE
Render dropdown content in document body

### PR DESCRIPTION
**Pull request by mperrotti**
Re Prelim:
- ~Still needs tests to be updated~
- ~Need to test in mup-web~

#### Related issues
Fixes https://meetup.atlassian.net/browse/WC-152

#### Description
Dropdown content needs to be rendered in the document body to avoid issues with `z-index` and `overflow`

### CODE REVIEW

**Comment by mperrotti** [comment]
Doing a more thorough check in `mup-web`, but feel free to review

File: assets/scss/components/_dropdown.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //
 $dropdownShadow: $interactiveShadow;
 $content-minWidth: $block-6;
-$content-maxWidth: 512px; // match chapstick
+// $content-maxWidth: 512px; // match chapstick
```
Comment (burrata) [inline assets/scss/components/_dropdown.scss]: should we remove this?

File: assets/scss/components/_dropdown.scss [hunk 2]
```diff
@@ -29,11 +29,9 @@ $spaceFromTrigger: $space-3 + $triangleSize;
 	background-color: $C_contentBG;
 	box-shadow: $dropdownShadow;
 	box-sizing: border-box;
+	margin-top: $triangleSize;
 	position: absolute;
 	top: $spaceFromTrigger;
-	min-width: $content-minWidth;
```
Comment (burrata) [inline assets/scss/components/_dropdown.scss]: 🙌  there's some `min-width: auto`s in mup-web that we can take out now.

File: src/interactive/Dropdown.jsx [hunk 3]
```diff
@@ -72,11 +106,15 @@ class Dropdown extends React.PureComponent {
 	componentDidMount() {
 		document.body.addEventListener('click', this.onBodyClick);
 		document.body.addEventListener('keydown', this.onBodyKeyDown);
+		window.addEventListener('resize', this.getContentPosition);
+		document.addEventListener('scroll', this.getContentPosition, true);
```
Comment (mperrotti) [inline src/interactive/Dropdown.jsx]: Might be good to figure out how to do this only when the dropdown is opened

File: src/interactive/Dropdown.jsx [hunk 4]
```diff
@@ -11,14 +13,44 @@ class Dropdown extends React.PureComponent {
...
+				case 'center':
+					return left + (width/2);
+				default:
+					return left - (contentWidth - width);
+			}
+		};
+
+		const ddPosition = {
+			x: getCoordX(this.props.align),
+			y: window.scrollY + top + height
```
Comment (mperrotti) [inline src/interactive/Dropdown.jsx]: Note for myself: `window.scrollY` may not work in all browsers

File: src/interactive/Dropdown.jsx [hunk 5]
```diff
@@ -30,6 +62,8 @@ class Dropdown extends React.PureComponent {
 	}
 
 	toggleContent(e) {
+		this.getContentPosition(e);
```
Comment (mperrotti) [inline src/interactive/Dropdown.jsx]: Note for myself: `getContentPosition` doesn't take an arg

File: src/interactive/Dropdown.jsx [hunk 6]
```diff
@@ -133,25 +172,38 @@ class Dropdown extends React.PureComponent {
...
-				</div>
+				<Portal isOpened={isActive}>
+					<div
+						ref={(el) => this.contentRef = el}
+						className={classNames.content}
+						aria-hidden={!isActive}
+						style={{
+							left: `${this.state.posX}px`,
+							top: `${this.state.posY}px`,
+							minWidth: `${width}px`,
```
Comment (burrata) [inline src/interactive/Dropdown.jsx]: Should these both be props and not include `px`? I can imagine wanting to have a separate `minWidth` and `maxWidth`, and for the values possibly to be `auto`.

File: src/interactive/Dropdown.jsx (hunk 3 above)
Comment (akdetrick) [reply src/interactive/Dropdown.jsx]: The listener is fine to leave in place while the component is mounted, but I'd recommend throttling, debouncing, or `requestAnimationFrame` on the `resize` and `scroll` handler functions

File: src/interactive/Dropdown.jsx [hunk 7]
```diff
@@ -133,25 +173,46 @@ class Dropdown extends React.PureComponent {
...
-				</div>
+				<Portal isOpened={isActive}>
+					<div
+						ref={(el) => this.contentRef = el}
+						className={classNames.content}
+						aria-hidden={!isActive}
+						style={{
+							left: `${this.state.posX}px`,
+							top: `${this.state.posY}px`,
+							minWidth: `${minWidth}`,
```
Comment (burrata) [inline src/interactive/Dropdown.jsx]: very minor, but this can now just be `minWidth` and the below prop `maxWidth`.

**Pull request merged**